fastlane/lib/fastlane/actions/appetize.rb

```diff
@@ -8,39 +8,63 @@ module SharedValues
     end
 
     class AppetizeAction < Action
-      APPETIZE_URL_BASE = 'https://api.appetize.io/v1/app/update'
-
       def self.is_supported?(platform)
         platform == :ios
       end
 
       def self.run(options)
         require 'net/http'
+        require 'net/http/post/multipart'
         require 'uri'
         require 'json'
 
-        uri = URI.parse(APPETIZE_URL_BASE)
-        https = Net::HTTP.new(uri.host, uri.port)
-        https.use_ssl = true
-
-        req = Net::HTTP::Post.new(uri.request_uri, initheader: {'Content-Type' => 'application/json'})
         params = {
-            token: options[:api_token],
-            url: options[:url],
             platform: 'ios'
         }
 
-        params[:privateKey] = options[:private_key] unless options[:private_key].nil?
-        req.body = JSON.generate(params)
-        response = https.request(req)
+        if options[:path]
+          params[:file] = UploadIO.new(options[:path], 'application/zip')
+        else
+          UI.user_error!('url parameter is required if no file path is specified') if options[:url].nil?
+          params[:url] = options[:url]
+        end
+
+        params[:note] = options[:note] unless options[:note].nil?
+
+        uri = URI.parse(appetize_url(options))
+        req = create_request(uri, params)
+        req.basic_auth(options[:api_token], nil)
 
-        UI.user_error!("Error when trying to upload ipa to Appetize.io") unless parse_response(response)
+        http = Net::HTTP.new(uri.host, uri.port)
+        http.use_ssl = true
+
+        response = http.request(req)
+
+        UI.user_error!("Error uploading app to Appetize.io") unless parse_response(response)
         UI.message("App URL: #{Actions.lane_context[SharedValues::APPETIZE_APP_URL]}")
         UI.message("Manage URL: #{Actions.lane_context[SharedValues::APPETIZE_MANAGE_URL]}")
         UI.message("App Private Key: #{Actions.lane_context[SharedValues::APPETIZE_PRIVATE_KEY]}")
         UI.success("Build successfully uploaded to Appetize.io")
       end
 
+      def self.appetize_url(options)
+        options[:public_key] ||= ""
```

Comment:
If the url does not contain a public key, then the app that's uploaded will be created as a new entry in appetize. If a public key is provided, it updates an existing entry, overwriting the old app with the new version.

As far as defaulting to "", I realize now that it isn't necessary. I had thought nil would become something else besides empty string during string interpolation, but quick test shows that's not the case, so I'll remove this line.